From 8fdcef23cf760a0d3bac22f2f95533736b4ba14e Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Tue, 2 Jun 2015 11:01:50 +0200
Subject: [PATCH 38/58] updater: disable updater pings

Despite auto-updater being arguably disabled (see previous commit),
Chromium would still send background requests. Kill it.
(trk:170, trk:171)
---
 .../component_updater/component_updater_url_constants.cc      | 4 ++--
 components/component_updater/configurator_impl.cc             | 3 ++-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/components/component_updater/component_updater_url_constants.cc b/components/component_updater/component_updater_url_constants.cc
index cd1b93217dc8..e600ca1a8a81 100644
--- a/components/component_updater/component_updater_url_constants.cc
+++ b/components/component_updater/component_updater_url_constants.cc
@@ -15,10 +15,10 @@ namespace component_updater {
 // The value of |kDefaultUrlSource| can be overridden with
 // --component-updater=url-source=someurl.
 const char kUpdaterDefaultUrl[] =
-    "https://update.googleapis.com/service/update2";
+    "trk:171:https://update.googleapis.com/service/update2";
 
 const char kUpdaterFallbackUrl[] =
-    "http://update.googleapis.com/service/update2";
+    "trk:171:http://update.googleapis.com/service/update2";
 
 const char kUpdaterJSONDefaultUrl[] =
     "https://update.googleapis.com/service/update2/json";
diff --git a/components/component_updater/configurator_impl.cc b/components/component_updater/configurator_impl.cc
index 2a964f77fa91..08125fb58d51 100644
--- a/components/component_updater/configurator_impl.cc
+++ b/components/component_updater/configurator_impl.cc
@@ -48,7 +48,7 @@ ConfiguratorImpl::ConfiguratorImpl(
     : background_downloads_enabled_(config_policy.BackgroundDownloadsEnabled()),
       deltas_enabled_(config_policy.DeltaUpdatesEnabled()),
       fast_update_(config_policy.FastUpdate()),
-      pings_enabled_(config_policy.PingsEnabled()),
+      pings_enabled_(false),
       require_encryption_(require_encryption),
       url_source_override_(config_policy.UrlSourceOverride()),
       initial_delay_(config_policy.InitialDelay()) {
@@ -81,6 +81,7 @@ int ConfiguratorImpl::UpdateDelay() const {
 std::vector<GURL> ConfiguratorImpl::UpdateUrl() const {
   if (url_source_override_.is_valid())
     return {GURL(url_source_override_)};
+  return {};
 
   std::vector<GURL> urls =
       base::FeatureList::IsEnabled(kFeatureUpdateClientUseJSON)
-- 
2.21.0

